RefrigeratorSaraFarkash/RefrigeratorSaraFarkash/CRefrigerator.cs

```diff
@@ -0,0 +1,334 @@
+                        this.shelfs[i].AddItem(item1);
+                        Console.WriteLine("the item is added!!!");
+                        break;
+                    }
+                }
+            }
+        }
+
+        //4
+        public CItem removeItemFromRef(int idItem)
+        {
+            CItem itemRemove;
+            for (int i = 0; i < this.numberOfShelves; i++)
+            {
+                if (shelfs[i].findItemInShelf(idItem))
+                {
+                    itemRemove = shelfs[i].removeItemfromShelf(idItem);
+                    return itemRemove;
+                }
+            }
+
+            Console.WriteLine("the item not found!!!");
+            return null;
+        }
+        //5
+        public void cleanExpired()
+        {
+            for (int i = 0; i < shelfs.Count; i++)
+            {
+                List<CItem> itemsToRemove = new List<CItem>();
+                foreach (CItem item in shelfs[i].Items)
+                {
+                    if (item.ExpiryDate < DateTime.Today)
+                    {
+                        Console.WriteLine("the product is" + item.ToString()
+                        + "Thrown in the trash due to expiration");
+                        itemsToRemove.Add(item);
+                    }
+                }
+                if (itemsToRemove.Count >= 0)
+                {
+                    // Since it is impossible to delete while in a loop, I created a list that stores all the expired products.
+                    foreach (CItem item in itemsToRemove)
+                    {
+                        CItem itemremoving = removeItemFromRef(item.Id);
+                    }
+                }
+            }
+        }
+
+        //6
+        public List<CItem> wantEat(string kashroot, string typeeat)
+        {
+            List<CItem> wanttoeat = new List<CItem>();
+
+            for (int i = 0; i < shelfs.Count; i++)
+            {
+                foreach (CItem item in shelfs[i].Items)
+                {
+                    if (item.Kashroot == kashroot && item.TypeI == typeeat && item.ExpiryDate > DateTime.Today)
+                        wanttoeat.Add(item);
+                }
+            }
+            return wanttoeat;
+        }
+
+        // sort
+
+        public int CompareTo(CRefrigerator otherRefrigerator)
+        {
+            return ((this.spaceLeftInRefrigerator() > otherRefrigerator.spaceLeftInRefrigerator()) ? (-1) : (this.spaceLeftInRefrigerator() == otherRefrigerator.spaceLeftInRefrigerator()) ? 0 : 1);
+        }
+
+        public List<CItem> SortItems()
+        {
+            List<CItem> items = new List<CItem>();
+            foreach (CShelf shelf in this.shelfs)
+            {
+                items.AddRange(shelf.Items);
+            }
+            items.Sort();
+            return items;
+        }
+        public List<CShelf> SortShelf()
+        {
+            List<CShelf> shelfs = new List<CShelf>();
```

Comment:
your are correct in not wanting to return the same list.
However you should have just used: List<CShelf>  shelfs =  new List<CShelf>(self.shelfs)